packages/vercel/src/index.ts

```diff
@@ -0,0 +1,620 @@
+							await builder.buildServerlessFolder(entryFile, NODE_PATH, _config.root);
+							if (isrConfig.exclude?.length) {
+								const dest = _middlewareEntryPoint ? MIDDLEWARE_PATH : NODE_PATH;
+								for (const route of isrConfig.exclude) {
+									// vercel interprets src as a regex pattern, so we need to escape it
+									routeDefinitions.push({ src: escapeRegex(route), dest });
+								}
+							}
+							await builder.buildISRFolder(entryFile, '_isr', isrConfig, _config.root);
+							for (const route of routes) {
+								const src = route.pattern.source;
+								const dest = src.startsWith('^\\/_image') ? NODE_PATH : ISR_PATH;
+								if (!route.prerender) routeDefinitions.push({ src, dest });
+							}
+						} else {
+							await builder.buildServerlessFolder(entryFile, NODE_PATH, _config.root);
+							const dest = _middlewareEntryPoint ? MIDDLEWARE_PATH : NODE_PATH;
+							for (const route of routes) {
+								if (!route.prerender) routeDefinitions.push({ src: route.pattern.source, dest });
+							}
+						}
+					}
+					if (_middlewareEntryPoint) {
+						await builder.buildMiddlewareFolder(
+							_middlewareEntryPoint,
+							MIDDLEWARE_PATH,
+							middlewareSecret
+						);
+					}
+				}
+				const fourOhFourRoute = routes.find((route) => route.pathname === '/404');
+				const destination =
+					buildOutput === 'server'
+						? new URL('./config.json', _config.outDir)
+						: new URL('./.vercel/output/config.json', _config.root);
+				const finalRoutes = [
+					...getRedirects(routes, _config),
+					{
+						src: `^/${_config.build.assets}/(.*)$`,
+						headers: { 'cache-control': 'public, max-age=31536000, immutable' },
+						continue: true,
+					},
+					{ handle: 'filesystem' },
+				];
+				if (buildOutput === 'server') {
+					finalRoutes.push(...routeDefinitions);
+				}
+
+				if (fourOhFourRoute) {
+					if (buildOutput === 'server') {
+						finalRoutes.push({
+							src: '/.*',
+							dest: fourOhFourRoute.prerender
+								? '/404.html'
+								: _middlewareEntryPoint
+									? MIDDLEWARE_PATH
+									: NODE_PATH,
+							status: 404,
+						});
+					} else {
+						finalRoutes.push({
+							src: '/.*',
+							dest: '/404.html',
+							status: 404,
+						});
+					}
+				}
+
+				let images: VercelImageConfig | undefined;
+				if (imageService || imagesConfig) {
+					if (imagesConfig) {
+						images = {
+							...imagesConfig,
+							domains: [...imagesConfig.domains, ..._config.image.domains],
+							remotePatterns: [
+								// ...(imagesConfig.remotePatterns ?? []),
+								// ..._config.image.remotePatterns,
+							],
+						};
+						if (imagesConfig.remotePatterns) {
+							images.remotePatterns?.push(...imagesConfig.remotePatterns);
+						}
+						const remotePatterns = _config.image.remotePatterns;
+						if (isAcceptedPattern(remotePatterns)) {
+							images.remotePatterns?.push(remotePatterns);
+						}
```

Comment:
Since `_config.image.remotePatterns` is an array, I think we need to filter the array here instead

Reply:
I am unsure I understand `_config.image.remotePatterns` **isn't** an array. That's what TS is telling me

Reply:
It shows an array for me though 🤔  

<img width="395" alt="image" src="https://github.com/user-attachments/assets/bf0f3966-c094-43e2-9a9d-ce051fa53cf2">

---
<img width="559" alt="image" src="https://github.com/user-attachments/assets/9c4fbd47-e1d1-4f8c-8d00-4a50fe27d0eb">

Reply:
AH!! I completely ignored the `[]` at the end... -.-